pkg/inject/inject.go

```diff
@@ -939,5 +949,22 @@ func (conf *ResourceConfig) InjectNamespace(annotations map[string]string) ([]by
 			ns.Annotations[annotation] = value
 		}
 	}
-	return yaml.Marshal(ns)
+
+	j, err := getFilteredJSON(ns)
+	if err != nil {
+		return nil, err
+	}
+	return yaml.JSONToYAML(j)
+}
+
+//getFilteredJSON method performs JSON marshaling such that zero values of
+//empty structs are respected by `omitempty` tags. We make use of a drop-in
+//replacement of the standard json/encoding library, without which empty struct values
+//present in workload objects would make it into the marshaled JSON.
+func getFilteredJSON(conf runtime.Object) ([]byte, error) {
```

Comment:
I think the `getFilteredJSON` method you have have here is identical to `jsonfilter.Marshal`.  It's great that we have this comment, but I don't think the function gives us any benefit.

Reply:
@adleong We'd have to be making use of the `jsonfilter.Marshal` method at multiple places. I wasn't really sure if it would be a nice idea to add such a comment at multiple places. I thought it would be a great idea to just have a reusable function that makes use of `jsonfilter.Marshal` that contributors could use in the future, instead of having to directly deal with a different version of `jso/encoding` by themselves, preserving the modularity of the codebase. 

WDYT?

Reply:
Gotcha.  In that case, I think the `getFilteredJSON` function body can simply be a one-liner: `return jsonfilter.Marshal(&conf)`